src/com/interview/dynamic/RegexMatching.java

```diff
@@ -52,14 +52,14 @@ public boolean matchRegex(char[] text, char[] pattern) {
 
         T[0][0] = true;
         //Deals with patterns like a* or a*b* or a*b*c*
-        for (int i = 1; i < T[0].length; i++) {
+        for (int i = 1; i <= T[0].length; i++) {
             if (pattern[i-1] == '*') {
                 T[0][i] = T[0][i - 2];
```

Comment:
for i=1, T[0][i - 2] column will be negative. Shouldn't it be T[0][i] = T[0][i - 1]?

Reply:
In below codes also i, j are starting from 1 and checks for pattern[j-2] and t[i][j-2] are used which would give error.

Reply:
In below codes also i, j are starting from 1 and checks for pattern[j-2] and t[i][j-2] are used which would give error